dial_worker.go

```diff
@@ -57,9 +57,10 @@ type dialWorker struct {
 	nextDial    []ma.Multiaddr
 	triggerDial <-chan struct{}
 
+	complete <-chan struct{}
```

Comment:
Can we have a comment what this chan does?

Reply:
sure!

Reply:
done.